### PR TITLE
Fix component page

### PR DESCRIPTION
**Pull request by franckgaudin**
The component pages were generating an error when data was passed to the props in the `.mdx` files. To solve this problem, we have set up a scope system which allows data to be passed to the components used in the files. 

From now on, you need to add the data you wish to pass to the component in the `contentConfig` file.

### CODE REVIEW

**Comment by fraincs** [comment]
Could installMethod become a component?

File: apps/docs/content/components/data.jsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+export const installMethods = [
```
Comment (fraincs) [inline apps/docs/content/components/data.jsx:1]: I can see this file growing with time could this be act as an index loading diverse data?

Comment (franckgaudin) [reply apps/docs/content/components/data.jsx:1]: Or you can create an index to load the data if it grows.

**Comment by franckgaudin** [comment]
> Could installMethod become a component?

we can make it evolve in the future yes

File: apps/docs/content/components/installation.mdx [hunk 2]
```diff
@@ -3,4 +3,26 @@ title: Installation
 description: The installation proccess
 ---
 
+import { installMethods } from "./data";
+
 # Installation
+
+## Install packages
+
+<Tabs tabs={installMethods}>
```
Comment (alexasselin008) [inline apps/docs/content/components/installation.mdx:12]: If this gets reused a couple of times, we could extract a component that takes an array of packages + a scope

PackageManagerInstallation with props packages=["@hopper-ui/components"] and scope="prod"

and the pnpm/yarn/npm functions would be synced automatically. 

I already see in this example that PNPM saves to prod, but npm and yarn save to dev which is probably an issue

Comment (franckgaudin) [reply apps/docs/content/components/installation.mdx]: Done

File: apps/docs/components/packageInstallation/PackageInstallation.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import { HighlightCode, highlightCode } from "@/components/highlightCode";
+
+import type { ReactElement } from "react";
+
+export interface PackageInstallationProps {
+    library: string;
+    mode: "dev" | "prod";
+}
+
+interface MethodIcons {
```
Comment (alexasselin008) [inline apps/docs/components/packageInstallation/PackageInstallation.tsx:11]: pour ton info, 

```
const methods = ["pnpm", "yarn", "npm"] as const;
type MethodIcons = Record<typeof methods[number], ReactElement>
```

Comment (alexasselin008) [reply apps/docs/components/packageInstallation/PackageInstallation.tsx:11]: ça t'éviterais un cast plus bas

Comment (franckgaudin) [reply apps/docs/components/packageInstallation/PackageInstallation.tsx]: Super merci pour l'info, l'update est fait

**Pull request merged**